This file is part of MXE.
See index.html for further information.

Taken from https://git.gnome.org/browse/glibmm/commit/?id=fb0bf14a29b80a0843e1e5dd1b5c5345d3839c93

From a53dd20596df824423088d70aa2c3ca262346cd5 Mon Sep 17 00:00:00 2001
From: Kjell Ahlstedt <kjell.ahlstedt@bredband.net>
Date: Tue, 2 Sep 2014 20:12:21 -0700
Subject: [PATCH] Gio::DBus: Don't use parameter name 'interface'

* gio/src/dbusmessage.hg:
* gio/src/dbusobject.hg: Change parameter name 'interface' to 'iface',
to avoid compilation errors with MinGW. Bug #735137.

Signed-off-by: Timothy Gu <timothygu99@gmail.com>

diff --git a/gio/src/dbusmessage.hg b/gio/src/dbusmessage.hg
index 3ca2511..926e7fb 100644
--- a/gio/src/dbusmessage.hg
+++ b/gio/src/dbusmessage.hg
@@ -1,5 +1,3 @@
-// -*- Mode: C++; indent-tabs-mode: nil; c-basic-offset: 2 -*-
-
 /* Copyright (C) 2010 The giomm Development Team
  *
  * This library is free software; you can redistribute it and/or
@@ -67,8 +65,11 @@ public:
   // They are generally UTF-8 that can only contain the ASCII subset,
   // so ustring still seems preferrable to std::string, which generally implies an unknown endoding for us.
 
-  _WRAP_METHOD(static Glib::RefPtr<Message> create_signal(const Glib::ustring& path, const Glib::ustring& interface, const Glib::ustring& signal), g_dbus_message_new_signal)
-  _WRAP_METHOD(static Glib::RefPtr<Message> create_method_call(const Glib::ustring& name, const Glib::ustring& path, const Glib::ustring& interface, const Glib::ustring& method), g_dbus_message_new_method_call)
+  // The parameter name 'interface' can cause compilation errors with MinGW.
+  // See https://bugzilla.gnome.org/show_bug.cgi?id=735137
+  // The parameter name in glib is 'interface_'.
+  _WRAP_METHOD(static Glib::RefPtr<Message> create_signal(const Glib::ustring& path, const Glib::ustring& iface, const Glib::ustring& signal), g_dbus_message_new_signal)
+  _WRAP_METHOD(static Glib::RefPtr<Message> create_method_call(const Glib::ustring& name, const Glib::ustring& path, const Glib::ustring& iface, const Glib::ustring& method), g_dbus_message_new_method_call)
   _WRAP_METHOD(static Glib::RefPtr<Message> create_method_reply(const Glib::RefPtr<Message>& method_call_message), g_dbus_message_new_method_reply)
   _WRAP_METHOD(static Glib::RefPtr<Message> create_method_error_literal(const Glib::RefPtr<const Message>& method_call_message, const Glib::ustring& error_name, const Glib::ustring& error_message), g_dbus_message_new_method_error_literal)
 
diff --git a/gio/src/dbusobject.hg b/gio/src/dbusobject.hg
index a1e47d6..5431004 100644
--- a/gio/src/dbusobject.hg
+++ b/gio/src/dbusobject.hg
@@ -1,5 +1,3 @@
-// -*- Mode: C++; indent-tabs-mode: nil; c-basic-offset: 2 -*-
-
 /* Copyright (C) 2012 The giomm Development Team
  *
  * This library is free software; you can redistribute it and/or
@@ -62,8 +60,11 @@ public:
 
 #m4 _CONVERSION(`GDBusInterface*',`const Glib::RefPtr<Gio::DBus::Interface>&',`Glib::wrap($3, true)')
 
-  _WRAP_SIGNAL(void interface_added(const Glib::RefPtr<Gio::DBus::Interface>& interface), "interface_added")
-  _WRAP_SIGNAL(void interface_removed(const Glib::RefPtr<Gio::DBus::Interface>& interface), "interface_removed")
+  // The parameter name 'interface' can cause compilation errors with MinGW.
+  // See https://bugzilla.gnome.org/show_bug.cgi?id=735137
+  // The parameter name in glib is 'interface_'.
+  _WRAP_SIGNAL(void interface_added(const Glib::RefPtr<Gio::DBus::Interface>& iface), "interface_added")
+  _WRAP_SIGNAL(void interface_removed(const Glib::RefPtr<Gio::DBus::Interface>& iface), "interface_removed")
 
 #m4 _CONVERSION(`Glib::ustring',`const gchar*',`g_strdup($3.c_str())')
   _WRAP_VFUNC(Glib::ustring get_object_path() const, "get_object_path")
-- 
1.8.3.2

